frontend-react/src/components/FileHandlers/FileHandler.test.tsx

```diff
@@ -577,5 +618,130 @@ describe("FileHandler", () => {
                 type: FileHandlerActionType.RESET,
             });
         });
+
+        describe("trackEvent", () => {
+            test("gets called on submit if FileHandlerType is type VALIDATION", async () => {
+                mockUseSenderResource.mockReturnValue({
```

Comment:
Some of this setup stuff could be pulled into a common function, like `renderComponent`:

```js
function renderComponent(fileHandlerType) {
    mockUserSenderResource.mockReturnValue(...);
    // ...other stuff
    renderWithFullAppContext(
        <FileHandler
            handlerType={fileHandlerType}
            /* other props */
         />
    );
}
```

Then just use that for each of the tests:

```js
describe("trackEvent", () => {
    test("gets called on submit if FileHandlerType is type VALIDATION", async () => {
         renderComponent(FileHandlerType.VALIDATION);

         // run validation expectations
    });

    test("gets called on submit if FileHandlerType is type UPLOAD", async () => {
         renderComponent(FileHandlerType.UPLOAD);

         // run upload/send expectations
    });
});
```

I also generally think it's better to keep test setup in `beforeEach`/`beforeAll` hooks, but definitely not a blocker.  None of this is a blocker at all, actually 😆